Sources/AppStoreConnectCLI/Commands/TestFlight/Sync/TestFlightProgramDifference.swift

```diff
@@ -0,0 +1,77 @@
+// Copyright 2020 Itty Bitty Apps Pty Ltd
+
+import Foundation
+import Model
+
+struct TestFlightProgramDifference {
+
+    enum Change {
+        case addBetaGroup(BetaGroup)
+        case removeBetaGroup(BetaGroup)
+        case addBetaTester(BetaTester, toBetaGroups: [BetaGroup])
+        case removeBetaTester(BetaTester, fromBetaGroups: [BetaGroup])
+
+        var description: String {
+
+            let operation: String = {
+                switch self {
+                case .addBetaGroup, .addBetaTester: return "added to"
+                case .removeBetaGroup, .removeBetaTester: return "removed from"
+                }
+            }()
+
+            switch self {
+            case .addBetaGroup(let betaGroup), .removeBetaGroup(let betaGroup):
+                let name = betaGroup.groupName ?? ""
+                let bundleId = betaGroup.app?.bundleId ?? ""
+
+                return "Beta group named: \(name) will be \(operation) app with bundleId: \(bundleId)"
+
+            case .addBetaTester(let betaTester, let betaGroups),
+                 .removeBetaTester(let betaTester, let betaGroups):
+                let email = betaTester.email ?? ""
+                let groupNames = betaGroups.map({ $0.groupName ?? "" }).joined(separator: ", ")
+
+                return "Beta Tester with email: \(email) will be \(operation) groups: \(groupNames)"
+            }
+        }
+    }
+
+    let changes: [Change]
+
+    init(local: TestFlightProgram, remote: TestFlightProgram) {
+        var changes: [Change] = []
+
+        // Groups
+        let localGroups = local.groups
+        let groupsToAdd: [BetaGroup] = localGroups.filter { $0.id == nil }
+        changes += groupsToAdd.map(Change.addBetaGroup)
+
+        let localGroupIds = localGroups.map(\.id)
+        let groupsToRemove: [BetaGroup] = remote.groups
+            .filter { group in localGroupIds.contains(group.id) == false }
+        changes += groupsToRemove.map(Change.removeBetaGroup)
+
+        // Testers
+        let newTesters = local.testers.filter { !remote.testers.map(\.email).contains($0.email) }
```

Comment:
After some testing, I find that if a tester is already in Group A, when you also want to add it to group B, this function can't detect that.
 
There might be a scennario missing here? A tester is in server, but the group in that tester is different.
```suggestion
        let newTesters = local.testers.filter { 
            if !remote.testers.map(\.email).contains(tester.email) { return true }
            else if remote.testers.map(\.email).contains(tester.email) {
                // Check if they have the same group, If not return true
            }
            return false
        }
```

Reply:
@DechengMa can you please post an example so that I can try it myself?
Also this `newTesters`section was meant to find testers that don't currently exist. So I would want to modify the section of code underneath this I think

Reply:
@DechengMa I believe I have fixed this edge case